proxy/src/main/java/com/velocitypowered/proxy/connection/client/ConnectedPlayer.java

```diff
@@ -695,8 +695,12 @@ void teardown() {
 
     DisconnectEvent.LoginStatus status;
     if (connectedPlayer.isPresent()) {
-      status = connectedPlayer.get() == this ? LoginStatus.SUCCESSFUL_LOGIN
-          : LoginStatus.CONFLICTING_LOGIN;
+      if (!connectedPlayer.get().getCurrentServer().isPresent()) {
```

Comment:
Maybe make connectedPlayer.get() to a variable and use that instead of explicitly calling .get

Reply:
Fixed in 4adc859cf11c62319f637d090bf0bc8d11f23ae8